src/BuildScriptGenerator.Common/ShellScriptBuilder.cs

```diff
@@ -161,23 +178,38 @@ public ShellScriptBuilder AddStringDoesNotExistInFileCheck(string searchString,
         /// <summary>
         /// Append a command to the shell script that sets the ORYX_SDK_STORAGE_BASE_URL to the value
         /// of ORYX_TEST_SDK_STORAGE_URL if ORYX_TEST_SDK_STORAGE_URL exists in the environment that is executing this code.
-        /// Otherwise, use the Oryx dev sdk storage account for testing.
+        /// Otherwise, use the Oryx staging private dev sdk storage account for testing.
         /// This allows us to change the storage account that tests use without regenerating any images.
         /// </summary>
         public ShellScriptBuilder AddDefaultTestEnvironmentVariables()
         {
             var testStorageAccountUrl = Environment.GetEnvironmentVariable(SdkStorageConstants.TestingSdkStorageUrlKeyName);
+            var sdkStorageUrl = string.IsNullOrEmpty(testStorageAccountUrl) ? SdkStorageConstants.PrivateStagingSdkStorageBaseUrl : testStorageAccountUrl;
 
-            return testStorageAccountUrl != null
-                ? this.SetEnvironmentVariable(SdkStorageConstants.SdkStorageBaseUrlKeyName, testStorageAccountUrl)
-                : this.SetEnvironmentVariable(SdkStorageConstants.SdkStorageBaseUrlKeyName, SdkStorageConstants.DevSdkStorageBaseUrl);
+            this.SetEnvironmentVariable(SdkStorageConstants.SdkStorageBaseUrlKeyName, sdkStorageUrl);
+            if (sdkStorageUrl == SdkStorageConstants.PrivateStagingSdkStorageBaseUrl)
+            {
+                 string stagingStorageSasToken = Environment.GetEnvironmentVariable(SdkStorageConstants.PrivateStagingStorageSasTokenKey) != null
+                   ? Environment.GetEnvironmentVariable(SdkStorageConstants.PrivateStagingStorageSasTokenKey)
+                    : this.GetKeyvaultSecretValue(SdkStorageConstants.OryxKeyvaultUri, SdkStorageConstants.StagingStorageSasTokenKeyvaultSecretName);
+                 this.SetEnvironmentVariable(SdkStorageConstants.PrivateStagingStorageSasTokenKey, stagingStorageSasToken);
+            }
+
+            return this;
         }
 
         public override string ToString()
         {
             return this.scriptBuilder.ToString();
         }
 
+        protected string GetKeyvaultSecretValue(string keyvaultUri, string secretName)
+        {
+            var client = new SecretClient(new Uri(keyvaultUri), new DefaultAzureCredential());
```

Comment:
Nit: May need to rename this as well from `keyvaultUri` to `keyVaultUri`

Reply:
done